src/basho_bench_worker.erl

```diff
@@ -201,8 +207,13 @@ worker_idle_loop(State) ->
     Driver = State#state.driver,
     receive
         {init_driver, Caller} ->
-            %% Spin up the driver implementation
-            case catch(Driver:new(State#state.id)) of
+            %% Spin up the driver implementation, optionally support new approach of passing State
+            DriverNew = 
+                case State#state.api_pass_state of 
+                    false -> catch(Driver:new(State#state.id));
+                     _ -> catch(Driver:new(State#state.id, State))
+                end,
+            case DriverNew of 
```

Comment:
No need to indent the case statement on a new line if it's not going over 80 characters. I would do:

```
            DriverNew = case State#state.api_pass_state of 
                    false -> catch(Driver:new(State#state.id));
                     _ -> catch(Driver:new(State#state.id, State))
            end,
            case DriverNew of
```